### PR TITLE
Day thirteen

### PR DESCRIPTION
**Pull request by Talaspaev**

### CODE REVIEW

File: src/Components/Users/users.jsx [hunk 1]
```diff
@@ -1,44 +1,79 @@
...
 import style from './users.module.css';
 
-const users = (props) => {
-  if (props.users.length === 0) {
-    props.setUsers(usersArray);
+import defaultImg from '../Image/person.jpg';
+
+class Users extends React.Component {
+  componentDidMount() {
+    axios.get(`https://social-network.samuraijs.com/api/1.0/users?page=${this.props.currentPage}&count=${this.props.pageSize}`)
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:15]: Consider breaking down this API url into separate pieces (BASE_URL, USER_RESOURCE, etc.) and placing them into variable at the core/constants level.

File: src/Components/Users/users.jsx [hunk 2]
```diff
@@ -1,44 +1,79 @@
...
+      });
   }
-  const toggleButton = (followed, id) => (followed ? <button type="button" onClick={() => { props.unfollowUser(id); }}>{I18N.ENG.UNFOLLOW}</button>
-    : <button type="button" onClick={() => { props.followUser(id); }}>{I18N.ENG.FOLLOW}</button>);
 
-  const usersItem = ({
-    id, photoUrl, followed, fullname, status, location,
-  }) => (
-    <div key={id} className={style.item}>
+  render() {
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:22]: I think we need to have a discussion about the things you put into "render" method.

File: src/Components/Users/users.jsx (hunk 1 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx]: i've add

File: src/Components/Users/users.jsx (hunk 2 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx]: it was very useful discussion

File: src/Components/Users/users.jsx [hunk 3]
```diff
@@ -1,44 +1,101 @@
...
+
+    axios.get(`${API.BASE_URL}${API.USER_RESOURCE}${API.PAGE}${currentPage}${API.COUNT}${pageSize}`)
+      .then(({ data }) => {
+        const { items, totalCount } = data;
+        setUsers(items);
+        setTotalUsersCount(totalCount);
+        this.a();
+      });
+  }
+  a = () => {
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:27]: Consider using more meaningful names for variables.

File: src/Components/Users/users.jsx [hunk 4]
```diff
@@ -1,44 +1,101 @@
...
+      });
+  };
+  createSpans = () => {
+    const { currentPage } = this.props;
+
+    return (
       <div>
-        <img src={photoUrl} alt="" className={style.img} />
-        {toggleButton(followed, id)}
+        {this.state.pages.map((p,index) => (
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:50]: Consider putting this JSX into separate function "createSpan".
Then in "createSpans" function you'll have mapping for `this.state.pages` using "createSpan" as a map executor.

File: src/Components/Users/users.jsx [hunk 5]
```diff
@@ -1,44 +1,101 @@
...
-};
-
-export default users;
+    );
+  pagesCount = () => {
+    const { totalUsersCount, pageSize, } = this.props;
+
+    return Math.ceil(totalUsersCount / pageSize);
+  }
+  usersContainer = () => {return this.props.users.map(this.usersItem);};
```
Comment (Dmitry-White) [inline src/Components/Users/users.jsx:89]: Consider adding empty lines around class methods.
Additionally, consider using arrow function return shorthand.

File: src/Components/Users/users.jsx (hunk 3 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx]: Sure.I hurried

File: src/Components/Users/users.jsx (hunk 4 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx]: i've done it

File: src/Components/Users/users.jsx (hunk 5 above)
Comment (Talaspaev) [reply src/Components/Users/users.jsx]: sure

**Pull request merged**